### ISSUE TITLE
Add support for calibration exposures

### ISSUE THREAD

**Issue by dkirkby** [desihub/specsim#54]
Specsim should be able to simulate calibration exposures (ARC, FLAT).  This requires knowing the spectrum and treating it like sky for throughput (without any moon or extinction, etc).

The spectra are not (yet) in desimodel, but available at nersc in `$DESI_ROOT/spectro/templates/calib/v0.3/`.  Need to figure out units and whether arc variability needs to be modeled.

**Comment by dkirkby** [comment]
The updated config should look something like:
```
instrument:
  ...
  calibration:
    paths:
      arc: calibs/arc_spectrum.fits
      flat: calibs/flat_spectrum.fits
  ...
observation:
  ...
  type: arc   # arc | flat | science
  ...
```

**Comment by sbailey** [comment]
This is now a blocking factor for 2% sprint so we need to get this implemented or otherwise work around it.

Rather than baking the spectra into desimodel and the specsim config, consider adding a `source_type="calib"` option that would apply all terms except moon and sky emission/absorption.  Input flux units from the user would be `erg / (s cm2 Angstrom arcsec2)` in that case.  This would give us the flexibility to also simulate effects like non-uniform lamp coverage, a burned out lamp, or the one-lamp-at-a-time calibration scheme described in [DESI-2761](https://desi.lbl.gov/DocDB/cgi-bin/private/ShowDocument?docid=2761).  We don't need that for the 2% sprint, but putting a single spectrum into the `instrument.calibration.paths.flat` config may not give us enough flexibility down the road.

**Comment by dkirkby** [comment]
I will take a look this afternoon.

**Comment by dkirkby** [comment]
Can you think of anything that would require arcs and flats to be simulated differently, other than the input surface brightness?

**Comment by sbailey** [comment]
In principle arcs and flats can be simulated the same other than the input surface brightness spectrum.

As a pragmatic matter, it can be handy to specify arcs as a set of delta functions at arbitrary wavelengths; otherwise you end up creating a super fine grid where most values are zero and the non-zero values are effectively not quite at the wavelength that you really wish they were.

Note: we've had discussions over the years about how valid is the approximation that arcs are delta functions, vs. slightly broadened lines with a small amount of continuum.

The current desisim + specter simulations directly use the ELECTRONS column rather than the CALIBRATED_FLUX column, which comes with these caveats in the header:
```
COMMENT ELECTRONS are electrons/s (typical exptime for BOSS=5s)                 
COMMENT CALIBRATED_FLUX are ergs/s/cm2/A which is wrong :                       
COMMENT - it is actually integrated lines (not per A)                           
COMMENT - one should remove BOSS atmosphere                                     
COMMENT - one should remove fiber aperture correction                           
COMMENT - RMS is large for one shoot of lamp to another                         
COMMENT - report the median of the exposures                                    
``` 
i.e. we'll need to revisit the input spectra; don't be too constrained by what is currently in those files but please do consider the delta function idea.  Otherwise we could still store the calib files as delta functions and convert them to a sampled surface brightness spectrum before passing it off to specsim.

**Comment by julienguy** [comment]
David, 
 * For arc lamps spectra, it would be good to use the actual expected line list for the DESI calibration system (the one used for now in sims is from BOSS). We did some measurements in Paris. You can find the list in  [docdb:2674](https://desi.lbl.gov/DocDB/cgi-bin/private/ShowDocument?docid=2674) (there is an ASCII file).
 * For continuum/flat, it would be good to add an extra 1% radial variation (from center to edge of field of view). This is what we expect to get with the calibration lamps. See [docdb:2761](https://desi.lbl.gov/DocDB/cgi-bin/private/ShowDocument?docid=2761) , figure 5.

**Comment by dkirkby** [comment]
Is it reasonable to assume that the plate scales calculated for sources at infinity are also valid for arcs and flats?  If not, is there a simple prescription I should use to correct the sky area of each fiber that multiplies the surface brightness?

**Comment by julienguy** [comment]
I think it is reasonable.
But, on top of the plate scale and vignetting, there is the effect that the
dome screen illumination is not uniform, and this translates into a 1%
effect from the center to the edge of the f.o.v that would be interesting
to include in the simulations.



2017-04-14 15:30 GMT+02:00 David Kirkby <notifications@github.com>:

**Comment by dkirkby** [comment]
Fiber-to-fiber variations of the incident surface brightness are no problem but the caller is responsible for modeling them.  Vignetting should be modeled in specsim but isn't yet - there is a separate issue for this #56.

**Comment by sbailey** [comment]
I agree that the caller of specsim should be responsible for modeling any dome screen non-uniformity.  I will include that (at least as an option) in the refactored newexp code.  If specsim can handle surface brightness inputs (i.e. flux per area) with some switch for not-applying moon/sky, that would be sufficient.

**Comment by dkirkby** [comment]
It sounds like #61 should fit the bill.

**Comment by sbailey** [comment]
calibration exposure support implemented in #61; closing this issue.
